pandas/index.pyx

```diff
@@ -426,6 +426,67 @@ cdef class Int64Engine(IndexEngine):
 
         return result
 
+cdef class UInt64Engine(IndexEngine):
+
```

Comment:
we are going to need to simplify this boilerplate, either expanding some methods in the superclass, maybe using some tempita. I *hate* copy-pasting code.